### ISSUE TITLE
Track mixpanel events through Service UI

### ISSUE THREAD

**Issue by fbarl** [weaveworks/scope#2460]
As @davkal mentioned, it would be nice to get some Scope metrics into our Mixpanel. For privacy reasons, we are only interested in tracking the events when Scope is used through Weave Cloud, in which case we could simply re-use the `window.mixpanel` handle set in the Service UI. Two approaches come to my mind:

1. Forward the mixpanel context/helpers to Scope component explicitly, e.g. `<Scope />` -> `<Scope trackMixpanelEvent={trackMixpanelEvent} />` in the Service UI. Good React patterns would then oblige us to forward the mixpanel stuff all the way down to the sub-components where they would be used which seems like a lot of forwarding.
2. Use exactly the same helpers like `trackMixpanelEvent` in the Scope UI and call them from there directly (as the `window.mixpanel` dependency will be available). The duplicated code wouldn't be that big and we could always extract it into the commons later on.

**Comment by davkal** [comment]
I'm fine with copying (and the check for a global `window.mixpanel`). Could yield some surprise events to someone else who integrates Scope as a component into their mixpanel-ed app 😉

**Comment by fbarl** [comment]
@jpellizzari pointed out to me that `process.env.WEAVE_CLOUD` could be used inside Scope to make sure we're inside Weave Cloud when using mixpanel, so everything should be fine with the second approach then!

**Comment by jpellizzari** [comment]
@fbarl The 'namespace' middleware now supports passing in a dependencies object that will be usable inside actions. It can be used to plumb 'services' through to actions; I already added a `registry` service to the dependencies:

https://github.com/weaveworks/service-ui/blob/master/client/src/main.jsx#L31

The registry is used by an action here:
https://github.com/weaveworks/service-ui/blob/master/client/src/actions/index.js#L695

You could create an action within Scope that will have access to that object, like so:

```javascript
export function trackMixpanelEvent(id, ...params) {
  return (dispatch, getState, deps) => {
    if (process.env.WEAVE_CLOUD && deps) {
      deps.mixpanel.someFunctionCall(id, params);
    }
  }
}
```
The mixpanel service would be created in service-ui and passed all the way through to Scope, avoiding the duplicate code.

**Comment by fbarl** [comment]
@jpellizzari Thanks for the tip, this looks very cool, except that I'm not sure if having a state action for handling mixpanel events is the best way. After all, this 'action' wouldn't really be modifying the state and I'm not sure if it will add any overhead to the computation of global state. Will try it out though!

**Comment by jpellizzari** [comment]
Since `dispatch` is never called, no state calculations happen. Its just a function call:
https://github.com/weaveworks/service-ui/blob/master/client/src/common/namespace.js#L53

Totally up to you, just wanted to mention that this way exists now.

**Comment by fbarl** [comment]
I was really not sure which approach to use in the end, so I talked to @davkal and @foot and we have agreed to go with a simple copying of `trackMixpanelEvent` helper.

Using the action with forwarded deps feels like a slightly overengineered solution to what we need at the moment, which is just one tiny helper from the `service-ui` code, but if we ever decide to bring in more mixpanel stuff into Scope, what you suggested might be a better way to go.

Good to know at least that the state recalculation would never be triggered in the example above! :)

**Comment by fbarl** [comment]
Addressed by #2462.
